cmake/TrMacros.cmake

```diff
@@ -196,17 +196,29 @@ function(tr_fixup_bundle_item BUNDLE_DIR BUNDLE_ITEMS DEP_DIRS)
 endfunction()
 
 macro(tr_qt_wrap_ui)
-    qt5_wrap_ui(${ARGN})
+    if(Qt_VERSION_MAJOR EQUAL 6)
+        qt6_wrap_ui(${ARGN})
+    else()
+        qt5_wrap_ui(${ARGN})
+    endif()
 endmacro()
 
 macro(tr_qt_add_resources)
-    qt5_add_resources(${ARGN})
+    if(Qt_VERSION_MAJOR EQUAL 6)
+        qt6_add_resources(${ARGN})
+    else()
+        qt5_add_resources(${ARGN})
+    endif()
 endmacro()
 
 macro(tr_qt_add_translation)
-    if (Qt5LinguistTools_VERSION VERSION_LESS 5.11.0)
-      qt5_add_translation(${ARGN})
+    if(Qt_VERSION GREATER_EQUAL 5.11)
+        if(Qt_VERSION_MAJOR EQUAL 6)
+            qt6_add_translation(${ARGN} OPTIONS -silent)
+        else()
+            qt5_add_translation(${ARGN} OPTIONS -silent)
+        endif()
     else()
-      qt5_add_translation(${ARGN} OPTIONS -silent)
+        qt5_add_translation(${ARGN})
```

Comment:
The nesting is unnecessary; just go in descending version order in a single depth

```
if (QtVERSION_MAJOR EQUAL 6)
    qt6_add_translation(${ARGN} OPTIONS -silent)
elseif (QtVERSION GREATER_EQUAL 5.11)
    qt5_add_translation(${ARGN} OPTIONS -silent)
else()
    qt5_add_translation(${ARGN})
endif()